### PR TITLE
Add Yarn Spinner

### PR DESCRIPTION
**Pull request by desplesda**
### SteamDB app page links to a few games using this

* [Night in the Woods](https://store.steampowered.com/app/481510/Night_in_the_Woods/)
* [Dredge](https://store.steampowered.com/app/1562430/DREDGE/)
* [Venba](https://store.steampowered.com/app/1491670/Venba/)
* [Lost in Random](https://store.steampowered.com/app/1462570/Lost_in_Random/)
* [A Short Hike](https://store.steampowered.com/app/1055540/A_Short_Hike/)

### Brief explanation of the change

This PR adds rules for detecting the Yarn Spinner narrative system, as well as tests and supporting files.

### CODE REVIEW

**Comment by SteamTracker** [comment]
After running changed rules in this pull request *(this bot only tests rules.ini changes)*:

Changed technologies: **SDK.YarnSpinner**

These apps now match: **(295)** [252390](https://steamdb.info/app/252390/depots/) [405710](https://steamdb.info/app/405710/depots/) [638990](https://steamdb.info/app/638990/depots/) [639660](https://steamdb.info/app/639660/depots/) [690100](https://steamdb.info/app/690100/depots/) [709710](https://steamdb.info/app/709710/depots/) [755170](https://steamdb.info/app/755170/depots/) [813540](https://steamdb.info/app/813540/depots/) [861930](https://steamdb.info/app/861930/depots/) [893710](https://steamdb.info/app/893710/depots/) [895760](https://steamdb.info/app/895760/depots/) [926610](https://steamdb.info/app/926610/depots/) [944330](https://steamdb.info/app/944330/depots/) [946770](https://steamdb.info/app/946770/depots/) [1001040](https://steamdb.info/app/1001040/depots/) [1014750](https://steamdb.info/app/1014750/depots/) [1015940](https://steamdb.info/app/1015940/depots/) [1033380](https://steamdb.info/app/1033380/depots/) [1035870](https://steamdb.info/app/1035870/depots/) [1096060](https://steamdb.info/app/1096060/depots/) [1140150](https://steamdb.info/app/1140150/depots/) [1183580](https://steamdb.info/app/1183580/depots/) [1189490](https://steamdb.info/app/1189490/depots/) [1197390](https://steamdb.info/app/1197390/depots/) [1199050](https://steamdb.info/app/1199050/depots/) [1221250](https://steamdb.info/app/1221250/depots/) [1265040](https://steamdb.info/app/1265040/depots/) [1272950](https://steamdb.info/app/1272950/depots/) [1275670](https://steamdb.info/app/1275670/depots/) [1276810](https://steamdb.info/app/1276810/depots/) [1281860](https://steamdb.info/app/1281860/depots/) [1290490](https://steamdb.info/app/1290490/depots/) [1298480](https://steamdb.info/app/1298480/depots/) [1301580](https://steamdb.info/app/1301580/depots/) [1302850](https://steamdb.info/app/1302850/depots/) [1302870](https://steamdb.info/app/1302870/depots/) [1304570](https://steamdb.info/app/1304570/depots/) [1305710](https://steamdb.info/app/1305710/depots/) [1309080](https://steamdb.info/app/1309080/depots/) [1310200](https://steamdb.info/app/1310200/depots/) [1316050](https://steamdb.info/app/1316050/depots/) [1318090](https://steamdb.info/app/1318090/depots/) [1320710](https://steamdb.info/app/1320710/depots/) [1322130](https://steamdb.info/app/1322130/depots/) [1325110](https://steamdb.info/app/1325110/depots/) [1325440](https://steamdb.info/app/1325440/depots/) [1325650](https://steamdb.info/app/1325650/depots/) [1325660](https://steamdb.info/app/1325660/depots/) [1330260](https://steamdb.info/app/1330260/depots/) [1330800](https://steamdb.info/app/1330800/depots/) [1332730](https://steamdb.info/app/1332730/depots/) [1334090](https://steamdb.info/app/1334090/depots/) [1334760](https://steamdb.info/app/1334760/depots/) [1335230](https://steamdb.info/app/1335230/depots/) [1339580](https://steamdb.info/app/1339580/depots/) [1354100](https://steamdb.info/app/1354100/depots/) [1361240](https://steamdb.info/app/1361240/depots/) [1371720](https://steamdb.info/app/1371720/depots/) [1394320](https://steamdb.info/app/1394320/depots/) [1398100](https://steamdb.info/app/1398100/depots/) [1404980](https://steamdb.info/app/1404980/depots/) [1406090](https://steamdb.info/app/1406090/depots/) [1411700](https://steamdb.info/app/1411700/depots/) [1421760](https://steamdb.info/app/1421760/depots/) [1427300](https://steamdb.info/app/1427300/depots/) [1432900](https://steamdb.info/app/1432900/depots/) [1433860](https://steamdb.info/app/1433860/depots/) [1434740](https://steamdb.info/app/1434740/depots/) [1435670](https://steamdb.info/app/1435670/depots/) [1440180](https://steamdb.info/app/1440180/depots/) [1440940](https://steamdb.info/app/1440940/depots/) [1444960](https://steamdb.info/app/1444960/depots/) [1447920](https://steamdb.info/app/1447920/depots/) [1450640](https://steamdb.info/app/1450640/depots/) [1466350](https://steamdb.info/app/1466350/depots/) [1491670](https://steamdb.info/app/1491670/depots/) [1493020](https://steamdb.info/app/1493020/depots/) [1503050](https://steamdb.info/app/1503050/depots/) [1504250](https://steamdb.info/app/1504250/depots/) [1508000](https://steamdb.info/app/1508000/depots/) [1525580](https://steamdb.info/app/1525580/depots/) [1529010](https://steamdb.info/app/1529010/depots/) [1529330](https://steamdb.info/app/1529330/depots/) [1530800](https://steamdb.info/app/1530800/depots/) [1534990](https://steamdb.info/app/1534990/depots/) [1536400](https://steamdb.info/app/1536400/depots/) [1543970](https://steamdb.info/app/1543970/depots/) [1544800](https://steamdb.info/app/1544800/depots/) [1545990](https://steamdb.info/app/1545990/depots/) [1552460](https://steamdb.info/app/1552460/depots/) [1557400](https://steamdb.info/app/1557400/depots/) [1558400](https://steamdb.info/app/1558400/depots/) [1562430](https://steamdb.info/app/1562430/depots/) [1566170](https://steamdb.info/app/1566170/depots/) [1574780](https://steamdb.info/app/1574780/depots/) [1576700](https://steamdb.info/app/1576700/depots/) [1577450](https://steamdb.info/app/1577450/depots/) [1580400](https://steamdb.info/app/1580400/depots/) [1590230](https://steamdb.info/app/1590230/depots/) [1592750](https://steamdb.info/app/1592750/depots/) [1601150](https://steamdb.info/app/1601150/depots/) [1612480](https://steamdb.info/app/1612480/depots/) [1613350](https://steamdb.info/app/1613350/depots/) [1618670](https://steamdb.info/app/1618670/depots/) [1622860](https://steamdb.info/app/1622860/depots/) [1631310](https://steamdb.info/app/1631310/depots/) [1632070](https://steamdb.info/app/1632070/depots/) [1646170](https://steamdb.info/app/1646170/depots/) [1660720](https://steamdb.info/app/1660720/depots/) [1663320](https://steamdb.info/app/1663320/depots/) [1663350](https://steamdb.info/app/1663350/depots/) [1667490](https://steamdb.info/app/1667490/depots/) [1670470](https://steamdb.info/app/1670470/depots/) [1673050](https://steamdb.info/app/1673050/depots/) [1677190](https://steamdb.info/app/1677190/depots/) [1678600](https://steamdb.info/app/1678600/depots/) [1678840](https://steamdb.info/app/1678840/depots/) [1682490](https://steamdb.info/app/1682490/depots/) [1689380](https://steamdb.info/app/1689380/depots/) [1692150](https://steamdb.info/app/1692150/depots/) [1702990](https://steamdb.info/app/1702990/depots/) [1703140](https://steamdb.info/app/1703140/depots/) [1721980](https://steamdb.info/app/1721980/depots/) [1734320](https://steamdb.info/app/1734320/depots/) [1738420](https://steamdb.info/app/1738420/depots/) [1740300](https://steamdb.info/app/1740300/depots/) [1743380](https://steamdb.info/app/1743380/depots/) [1745270](https://steamdb.info/app/1745270/depots/) [1763260](https://steamdb.info/app/1763260/depots/) [1771440](https://steamdb.info/app/1771440/depots/) [1778590](https://steamdb.info/app/1778590/depots/) [1788550](https://steamdb.info/app/1788550/depots/) [1800510](https://steamdb.info/app/1800510/depots/) [1803160](https://steamdb.info/app/1803160/depots/) [1807460](https://steamdb.info/app/1807460/depots/) [1808360](https://steamdb.info/app/1808360/depots/) [1812090](https://steamdb.info/app/1812090/depots/) [1821000](https://steamdb.info/app/1821000/depots/) [1838940](https://steamdb.info/app/1838940/depots/) [1839180](https://steamdb.info/app/1839180/depots/) [1849040](https://steamdb.info/app/1849040/depots/) [1865310](https://steamdb.info/app/1865310/depots/) [1869870](https://steamdb.info/app/1869870/depots/) [1873060](https://steamdb.info/app/1873060/depots/) [1873860](https://steamdb.info/app/1873860/depots/) [1878130](https://steamdb.info/app/1878130/depots/) [1882960](https://steamdb.info/app/1882960/depots/) [1890280](https://steamdb.info/app/1890280/depots/) [1917200](https://steamdb.info/app/1917200/depots/) [1924360](https://steamdb.info/app/1924360/depots/) [1924930](https://steamdb.info/app/1924930/depots/) [1928430](https://steamdb.info/app/1928430/depots/) [1942720](https://steamdb.info/app/1942720/depots/) [1961340](https://steamdb.info/app/1961340/depots/) [1962980](https://steamdb.info/app/1962980/depots/) [1965840](https://steamdb.info/app/1965840/depots/) [1975250](https://steamdb.info/app/1975250/depots/) [1987090](https://steamdb.info/app/1987090/depots/) [1988220](https://steamdb.info/app/1988220/depots/) [1989140](https://steamdb.info/app/1989140/depots/) [2000430](https://steamdb.info/app/2000430/depots/) [2001940](https://steamdb.info/app/2001940/depots/) [2004400](https://steamdb.info/app/2004400/depots/) [2005650](https://steamdb.info/app/2005650/depots/) [2018350](https://steamdb.info/app/2018350/depots/) [2019450](https://steamdb.info/app/2019450/depots/) [2026430](https://steamdb.info/app/2026430/depots/) [2027080](https://steamdb.info/app/2027080/depots/) [2053430](https://steamdb.info/app/2053430/depots/) [2057380](https://steamdb.info/app/2057380/depots/) [2058460](https://steamdb.info/app/2058460/depots/) [2075160](https://steamdb.info/app/2075160/depots/) [2075670](https://steamdb.info/app/2075670/depots/) [2078360](https://steamdb.info/app/2078360/depots/) [2079160](https://steamdb.info/app/2079160/depots/) [2079790](https://steamdb.info/app/2079790/depots/) [2080930](https://steamdb.info/app/2080930/depots/) [2087450](https://steamdb.info/app/2087450/depots/) [2089180](https://steamdb.info/app/2089180/depots/) [2089320](https://steamdb.info/app/2089320/depots/) [2098190](https://steamdb.info/app/2098190/depots/) [2101570](https://steamdb.info/app/2101570/depots/) [2106980](https://steamdb.info/app/2106980/depots/) [2117790](https://steamdb.info/app/2117790/depots/) [2123770](https://steamdb.info/app/2123770/depots/) [2138570](https://steamdb.info/app/2138570/depots/) [2149480](https://steamdb.info/app/2149480/depots/) [2158160](https://steamdb.info/app/2158160/depots/) [2160850](https://steamdb.info/app/2160850/depots/) [2163510](https://steamdb.info/app/2163510/depots/) [2164970](https://steamdb.info/app/2164970/depots/) [2166240](https://steamdb.info/app/2166240/depots/) [2175530](https://steamdb.info/app/2175530/depots/) [2176310](https://steamdb.info/app/2176310/depots/) [2179510](https://steamdb.info/app/2179510/depots/) [2182430](https://steamdb.info/app/2182430/depots/) [2193920](https://steamdb.info/app/2193920/depots/) [2208970](https://steamdb.info/app/2208970/depots/) [2209040](https://steamdb.info/app/2209040/depots/) [2212520](https://steamdb.info/app/2212520/depots/) [2222950](https://steamdb.info/app/2222950/depots/) [2226770](https://steamdb.info/app/2226770/depots/) [2226900](https://steamdb.info/app/2226900/depots/) [2243270](https://steamdb.info/app/2243270/depots/) [2243900](https://steamdb.info/app/2243900/depots/) [2260040](https://steamdb.info/app/2260040/depots/) [2269220](https://steamdb.info/app/2269220/depots/) [2269450](https://steamdb.info/app/2269450/depots/) [2270460](https://steamdb.info/app/2270460/depots/) [2283020](https://steamdb.info/app/2283020/depots/) [2290900](https://steamdb.info/app/2290900/depots/) [2291130](https://steamdb.info/app/2291130/depots/) [2291610](https://steamdb.info/app/2291610/depots/) [2297650](https://steamdb.info/app/2297650/depots/) [2299040](https://steamdb.info/app/2299040/depots/) [2304900](https://steamdb.info/app/2304900/depots/) [2309270](https://steamdb.info/app/2309270/depots/) [2315680](https://steamdb.info/app/2315680/depots/) [2317280](https://steamdb.info/app/2317280/depots/) [2337850](https://steamdb.info/app/2337850/depots/) [2348950](https://steamdb.info/app/2348950/depots/) [2364770](https://steamdb.info/app/2364770/depots/) [2372070](https://steamdb.info/app/2372070/depots/) [2379360](https://steamdb.info/app/2379360/depots/) [2380760](https://steamdb.info/app/2380760/depots/) [2382620](https://steamdb.info/app/2382620/depots/) [2393560](https://steamdb.info/app/2393560/depots/) [2398310](https://steamdb.info/app/2398310/depots/) [2402780](https://steamdb.info/app/2402780/depots/) [2403850](https://steamdb.info/app/2403850/depots/) [2408090](https://steamdb.info/app/2408090/depots/) [2410700](https://steamdb.info/app/2410700/depots/) [2411060](https://steamdb.info/app/2411060/depots/) [2421600](https://steamdb.info/app/2421600/depots/) [2422320](https://steamdb.info/app/2422320/depots/) [2422520](https://steamdb.info/app/2422520/depots/) [2426850](https://steamdb.info/app/2426850/depots/) [2429560](https://steamdb.info/app/2429560/depots/) [2463050](https://steamdb.info/app/2463050/depots/) [2464630](https://steamdb.info/app/2464630/depots/) [2469630](https://steamdb.info/app/2469630/depots/) [2481880](https://steamdb.info/app/2481880/depots/) [2487630](https://steamdb.info/app/2487630/depots/) [2493400](https://steamdb.info/app/2493400/depots/) [2493650](https://steamdb.info/app/2493650/depots/) [2497790](https://steamdb.info/app/2497790/depots/) [2500600](https://steamdb.info/app/2500600/depots/) [2506160](https://steamdb.info/app/2506160/depots/) [2506960](https://steamdb.info/app/2506960/depots/) [2506970](https://steamdb.info/app/2506970/depots/) [2509090](https://steamdb.info/app/2509090/depots/) [2513890](https://steamdb.info/app/2513890/depots/) [2522720](https://steamdb.info/app/2522720/depots/) [2532300](https://steamdb.info/app/2532300/depots/) [2535770](https://steamdb.info/app/2535770/depots/) [2546090](https://steamdb.info/app/2546090/depots/) [2553710](https://steamdb.info/app/2553710/depots/) [2555020](https://steamdb.info/app/2555020/depots/) [2556900](https://steamdb.info/app/2556900/depots/) [2561320](https://steamdb.info/app/2561320/depots/) [2563670](https://steamdb.info/app/2563670/depots/) [2568590](https://steamdb.info/app/2568590/depots/) [2573380](https://steamdb.info/app/2573380/depots/) [2575990](https://steamdb.info/app/2575990/depots/) [2589310](https://steamdb.info/app/2589310/depots/) [2597480](https://steamdb.info/app/2597480/depots/) [2598120](https://steamdb.info/app/2598120/depots/) [2600760](https://steamdb.info/app/2600760/depots/) [2602380](https://steamdb.info/app/2602380/depots/) [2602780](https://steamdb.info/app/2602780/depots/) [2603340](https://steamdb.info/app/2603340/depots/) [2611580](https://steamdb.info/app/2611580/depots/) [2629970](https://steamdb.info/app/2629970/depots/) [2636670](https://steamdb.info/app/2636670/depots/) [2636700](https://steamdb.info/app/2636700/depots/) [2657200](https://steamdb.info/app/2657200/depots/) [2671730](https://steamdb.info/app/2671730/depots/) [2671740](https://steamdb.info/app/2671740/depots/) [2680090](https://steamdb.info/app/2680090/depots/) [2685360](https://steamdb.info/app/2685360/depots/) [2686930](https://steamdb.info/app/2686930/depots/) [2704270](https://steamdb.info/app/2704270/depots/) [2721980](https://steamdb.info/app/2721980/depots/) [2742880](https://steamdb.info/app/2742880/depots/) [2750210](https://steamdb.info/app/2750210/depots/) [2756310](https://steamdb.info/app/2756310/depots/) [2796550](https://steamdb.info/app/2796550/depots/) [2799700](https://steamdb.info/app/2799700/depots/) [2800230](https://steamdb.info/app/2800230/depots/) [2801500](https://steamdb.info/app/2801500/depots/) [2835640](https://steamdb.info/app/2835640/depots/) [2846250](https://steamdb.info/app/2846250/depots/) [2883960](https://steamdb.info/app/2883960/depots/) [2890290](https://steamdb.info/app/2890290/depots/) [2892260](https://steamdb.info/app/2892260/depots/)

Please confirm that all these changed apps are correct.

File: rules.ini [hunk 1]
```diff
@@ -314,6 +314,7 @@ Vivox[] = vivox
 Vorbis = vorbis
 Wwise = (?:^|/)AkSoundEngine(?:dll)?\.(?:bundle|dll)$
 XAudio2 = (?:^|/)xaudio2_9redist\.dll$
+YarnSpinner = (?:^|/)[Yy]arn[Ss]pinner(?:\.[Cc]ompiler)?\.dll$
```
Comment (Lifeismana) [inline rules.ini:317]:
```suggestion
YarnSpinner = (?:^|/)YarnSpinner(?:\.Compiler)?\.dll$
```
not needed since the regex is ran in case insensitive mode

Comment (Lifeismana) [reply rules.ini:317]:
```suggestion
YarnSpinner = (?:^|/)YarnSpinner\.dll$
```
after looking into it, the compiler part is also not needed, there no app that have YarnSpinner.Compiler.dll without also having YarnSpinner.dll

Comment (desplesda) [reply rules.ini]: I've removed the case-sensitivity check, and removed the check for '.Compiler'.

Comment (Lifeismana) [reply rules.ini]: Nice, now the only thing left is to fix the test file (remove test with case sensitive & compiler in them)

Comment (desplesda) [reply rules.ini]: Done!

File: tests/types/SDK.YarnSpinner.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/YarnSpinner.dll
+/yarnspinner.dll
+YarnSpinner.dll
+yarnspinner.dll
```
Comment (Lifeismana) [inline tests/types/SDK.YarnSpinner.txt:4]:
```suggestion
/YarnSpinner.dll
YarnSpinner.dll
```

**Comment by Lifeismana** [comment]
Perfect, Thank you!

**Comment by desplesda** [comment]
Thank you very much!

**Pull request merged**